### PR TITLE
Add the examples where one would need IIFE patterns

### LINKED ISSUE #2958: Issue with "IIFE": example doesn’t actually show a case where an IIFE is necessary
MDN URL: https://developer.mozilla.org/en-US/docs/Glossary/IIFE

#### What information was incorrect, unhelpful, or incomplete?

(function () {
    var aName = "Barry";
})();
// Variable aName is not accessible from the outside scope
aName // throws "Uncaught ReferenceError: aName is not defined"

#### Specific section or headline?

https://developer.mozilla.org/en-US/docs/Glossary/IIFE#examples

#### What did you expect to see?

Achieving the same result with function declaration and function expression, so what make it special in IIFE.

function foo(){
	var aName="Barry";
};
foo(); //I called the function to replicate effect of IIFE that function was called.
// Variable aName is not accessible from the outside scope
aName // throws "Uncaught ReferenceError: aName is not defined"

var foo = function(){
	var aName="Barry";
};

foo();
// Variable aName is not accessible from the outside scope
aName // throws "Uncaught ReferenceError: aName is not defined"

#### Did you test this? If so, how?

Yes tested on raw js file.

<details>
<summary>MDN Content page report details</summary>

* Folder: `en-us/glossary/iife`
* MDN URL: https://developer.mozilla.org/en-US/docs/Glossary/IIFE
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/glossary/iife/index.html
* Last commit: https://github.com/mdn/content/commit/78db9a278d0b412b147c26df27eb3d843ba832e3
* Document last modified: 2020-12-30T11:27:15.000Z

</details>

### PR DESCRIPTION
**Pull request by longm89**
> What was wrong/why is this fix needed? (quick summary only)
Examples don't actually show a case where an IIFE is necessary

> MDN URL of main page changed
https://developer.mozilla.org/en-US/docs/Glossary/IIFE

> Issue number (if there is an associated issue)
Fixes #2958

> Anything else that could help us review it
I added some use cases for the IIFE patterns.

### CODE REVIEW

File: files/en-us/glossary/iife/index.html [hunk 1]
```diff
@@ -21,26 +23,97 @@
...
-<p>Assigning the IIFE to a variable stores the function's return value, not the function definition itself.</p>
-
-<pre class="brush: js">var result = (function () {
-    var name = "Barry";
-    return name;
-})();
-// Immediately creates the output:
-result; // "Barry"</pre>
+<h2 id="Examples">Some use cases</h2>
+  <h3>Avoid polluting the global namespace</h3>
```
Comment (chrisdavidmills) [inline files/en-us/glossary/iife/index.html:27]: From here on in, the spacing needs some attention. Make sure all lines are flush to the left of the file, and that code blocks use two spaces per indentation level as per MDN guidelines.

Comment (chrisdavidmills) [inline files/en-us/glossary/iife/index.html:27]: Headings need an ID, which should be the same as the heading text, except all in lower case, with underscores between each word.

Comment (chrisdavidmills) [reply files/en-us/glossary/iife/index.html:27]: So this one would be `avoid_polluting_the_global_namespace`

File: files/en-us/glossary/iife/index.html [hunk 2]
```diff
@@ -21,26 +23,97 @@
...
-
-<p>Assigning the IIFE to a variable stores the function's return value, not the function definition itself.</p>
-
-<pre class="brush: js">var result = (function () {
-    var name = "Barry";
-    return name;
-})();
-// Immediately creates the output:
-result; // "Barry"</pre>
+<h2 id="Examples">Some use cases</h2>
```
Comment (chrisdavidmills) [inline files/en-us/glossary/iife/index.html:26]: Here, the id doesn't match the heading wording. I'd go for `Use cases` / `use_cases`

File: files/en-us/glossary/iife/index.html (hunk 1 above)
Comment (chrisdavidmills) [reply files/en-us/glossary/iife/index.html:27]: Ther are other instances below too.

File: files/en-us/glossary/iife/index.html [hunk 3]
```diff
@@ -21,26 +23,97 @@
...
+      const secondAccount = makeWithdraw(20);
+      secondAccount.withdraw(30); // "Insufficient money"
+      secondAccount.withdraw(20);  // 0
+  </pre>
 
+  <h3>The for loop with var before ES6</h3>
+  <p>We could see the following pattern in old code, before the introduction of the statements <strong>let</strong> and <strong>const</strong> 
+    in <strong>ES6</strong> and the block scope. With the statement <strong>var</strong>, we have only function scopes and the global scope. 
+  </p>
+  Suppose we want to create 2 buttons with the text Button 0 and Button 1 and we click
```
Comment (chrisdavidmills) [inline files/en-us/glossary/iife/index.html:79]: From here, there are a few instances of paragraphs not wrapped inside `<p></p>`. Make sure you add those, as in the final rendered page the spacing is a bit off as a result.

**Review by chrisdavidmills — CHANGES_REQUESTED** [review]
Thanks for the addition @longm89 ! The content is really good; there are a just few comments for you to address related to cleaning up the file a bit.

**Comment by longm89** [comment]
@chrisdavidmills Thank you very much for the corrections. I fixed the code and header styles and pushed to the same local branch. It's all needed and I don't need to create a new pull request?

**Comment by chrisdavidmills** [comment]
@longm89 yes, you just push to the same remote and it automatically adds it to the pull request. You don't need to create a new pull request

**Review by chrisdavidmills — APPROVED** [review]
OK, I've just done a bit of copy edit/cleanup of the source, but I'm pretty happy with this now. Nice work @longm89 , thank you!

**Comment by longm89** [comment]
@chrisdavidmills Thank you very much for the changes!

**Pull request merged**